packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -312,7 +312,17 @@ class GlobalObjectKey<T extends State<StatefulWidget>> extends GlobalKey<T> {
   int get hashCode => identityHashCode(value);
 
   @override
-  String toString() => '[$runtimeType ${describeIdentity(value)}]';
+  String toString() {
+    String selfType = runtimeType.toString();
+    // This is a temporary work-around to align what const GlobalObjectKey().toString()
+    // produces in Dart 2 with what it produces in Dart 1.
+    // In Dart 2 const GlobalObjectKey() is instantiated to bounds so it becomes
+    // const GlobalObjectKey<State<StatefulWidget>>().
+    if (selfType == 'GlobalObjectKey<State<StatefulWidget>>') {
+      selfType = 'GlobalObjectKey';
+    }
```

Comment:
why is this only a temporary work-around?

Seems like we'll need something permanently for this, no?

Can we compare T instead of the reified string? That might be cleaner.

Reply:
I will remove the word temporary from my comment.

Comparison against `T` does not work because `GlobalObjectKey` can have subclasses, so if I do something like:

```dart
if (T == State) {
  selfType = 'GlobalObjectKey';
}
```

that would be incorrect for subclasses. 

Alternative to full matching I could do:

```dart
const suffix = '<State<StatefulWidget>>'
if (selfType.endsWith(suffix)) {
  selfType = selfType.substring(0, selfType.length - suffix.length);
}
```

What do you think?

Reply:
> I will remove the word temporary from my comment.

I would just remove all the mention of Dart 1, too. We're not doing this because of what Dart 1 did, we're doing it because we want a particular effect to make the dumps less confusing and cleaner, and that's how you get the effect.

> ...subclasses...

Ah, yeah. Your proposed code seems reasonable. We should test for that too if we don't already (and test for the case where we do want the output to include generics).

Reply:
Addressed comments, added test. Please take another look.